inst/include/dust/gpu/distr/binomial.hpp

```diff
@@ -0,0 +1,150 @@
+#ifndef DUST_DISTR_BINOMIAL_HPP
+#define DUST_DISTR_BINOMIAL_HPP
+
+#include <cmath>
+
+namespace dust {
+namespace distr {
+
+template <typename T>
+__device__
+double binomial_inversion(rng_state_t<T>& rng_state,
+                          double n, double prob) {
+  double geom_sum = 0;
+  double num_geom = 0;
+
+  while (true) {
+    double r = device_unif_rand(rng_state);
+    double geom = ceil(log(r) / log1p(-prob));
+    geom_sum += geom;
+    if (geom_sum > n) {
+      break;
+    }
+    ++num_geom;
+  }
+  __syncwarp(); // CHECK: ok to call this here, when some threads in warp
```

Comment:
We can remove this comment - empirically these `__syncwarp();` didn't deadlock

Reply:
done in 529ecde